configuration/src/main/java/io/airlift/configuration/ConfigurationFactory.java

```diff
@@ -546,6 +568,7 @@ private Object getInjectedValue(AttributeMetadata attribute, ConfigurationMetada
     {
         String name = prefix + injectionPoint.getProperty();
         usedProperties.add(name);
```

Comment:
Can we have a single list instead?

Reply:
Unfortunately not, as the older approach only stores property names and is not aware of whether a property is sensitive (see `ConfigurationFactory#consumeProperty(String)`).

Reply:
We can break compatibility here. There is a single use-case in Trino where we use it and it's based on values returned from other Bootstrap.